src/database.rs

```diff
@@ -39,8 +39,9 @@ impl ReadOnlyDB {
         }
 
         let mut code_db = CodeDB::new();
-        for (hash, code) in collect_codes(l2_trace, Some(&sdb)).unwrap() {
-            code_db.insert_with_hash(hash, code);
+        for code_trace in l2_trace.codes.iter() {
+            let hash = code_db.insert(code_trace.code.to_vec());
```

Comment:
This change is good/sound. While this time let us do not do this change. So we can compare Apple to Apple for cycles.